posts/2025-04-08-25.0.0.4-beta.adoc

```diff
@@ -0,0 +1,251 @@
+
+For more information on Java 24, see the following resources:
+
+- link:https://jdk.java.net/24/release-notes[Release notes]
+- link:https://download.java.net/java/early_access/jdk24/docs/api/[Javadocs]
+- link:https://jdk.java.net/24/[Download page].
+
+[#telemetry]
+== Send Liberty HTTP access logs to OpenTelemetry using MicroProfile Telemetry 2.0
+
+link:https://openliberty.io/docs/latest/microprofile-telemetry.html[MicroProfile Telemetry 2.0] (`mpTelemetry-2.0`) enables OpenTelemetry to collect and export traces, metrics, and logs.
+
+You can now use the `mpTelemetry-2.0` feature to collect link:https://openliberty.io/docs/latest/access-logging.html[Liberty HTTP access logs] to OpenTelemetry. This new log source is in addition to the previously available Open Liberty runtime log sources (messages, traces, ffdcs, audit) and application logs generated by the `java.util.logging` (JUL) API.
+
+To enable the MicroProfile Telemetry 2.0 feature to collect HTTP access logs, enable the feature and configure the `httpEndpoint` and optionally the `httpAccessLogging` configuration elements. You must also specify the new `accessLog` source to the source attribute for the `mpTelemetry` configuration element, as shown in the following `server.xml` example:
+
+[source,xml]
+----
+<featureManager>
+    <feature>mpTelemetry-2.0</feature>
+    ...
+</featureManager>
+
+<httpAccessLogging id="accessLogging" logFormat='%h %u %{t}W "%r" %s %b %D %{R}W'/>
+<httpEndpoint id="defaultHttpEndpoint" accessLoggingRef="accessLogging" httpPort="9080" httpsPort="9443"/>
+
+<mpTelemetry accessLogFields=“logFormat” source=“message,trace,ffdc,accessLog”/>
+----
+
+
+[#backport]
+== File-based health checks and Java EE 7 and 8 compatibility for MicroProfile Health 4.0
+
+The MicroProfile Health 4.0 feature (`mpHealth-4.0`) enables a runtime for the Open Liberty server that previously supported only Jakarta EE 9 or later. In this beta release, the `mpHealth-4.0` feature is now also supported with in Java EE7 and EE8 environments. Also, a file-based health check mechanism is introduced as an alternative to using the `/health/*` REST endpoints.
+
+The MicroProfile Health technology is most effective when the application is deployed onto a Kubernetes based environment where link:https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/['Liveness', 'Readiness' and 'Startup' Probes] can be used. In such a deployment scenario, applications might be deployed are on different Java and Jakarta EE levels. This configuration can result in a mixture of `mpHealth-x.x` features across different servers and different Kubernetes probe configurations. The `mpHealth-4.0` feature now provides technological concurrency and unifies Open Libety server configurations and Kubernetes health check probe configurations. This feature is now compatible to use with Java EE7 and Java EE8 alongside its existing support for Jakarta EE9 and later environments.
```

Comment:
"In such a deployment scenario, applications might be deployed are on different". Not sure if it's me, but that sounds a bit weird? should it be "application that might be deployed are on"? or "applications that are deployed might be on "

Also for "The `mpHealth-4.0` feature now provides technological concurrency". Maye sound a bit weird to say that the `mpHealth-4.0` is providing technological concurrency? In the original issue I meant to use that as users that are  using mpHealth-4.0 will be "up to date" with the latest tech (in regards to it also being compatible for EE8 and EE7 envs).